### PR TITLE
fix(cli): handle conflicting script names

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes https://linear.app/nango/issue/NAN-1674/cli-conflicting-script-name-is-not-handled-correctly

- Handle conflicting scripts names
It was selecting the first one instead of failed explicitly 

- Improve input when no provider matched
- Improve input when no script matched
- Bonus: fix sdk error message when records failed validation

Screenshot 2024-09-06 at 17 25 25
Screenshot 2024-09-06 at 17 25 47
Screenshot 2024-09-06 at 17 25 33

### CODE REVIEW

File: packages/cli/lib/services/dryrun.service.ts [hunk 1]
```diff
@@ -97,61 +97,72 @@ export class DryRunService {
             return;
         }
 
-        let providerConfigKey = options.optionalProviderConfigKey;
+        if (options.optionalProviderConfigKey && !response.parsed.integrations.some((inte) => inte.providerConfigKey === options.optionalProviderConfigKey)) {
+            console.log(chalk.red(`Integration "${options.optionalProviderConfigKey}" does not exists`));
```
Comment (khaliqgant) [inline packages/cli/lib/services/dryrun.service.ts]: does not exist

Comment (bodinsamuel) [reply packages/cli/lib/services/dryrun.service.ts]: every time 😮‍💨

**Review by khaliqgant — APPROVED** [review]

File: packages/cli/lib/services/dryrun.service.ts [hunk 2]
```diff
@@ -97,61 +97,72 @@ export class DryRunService {
...
+                    isPostConnectionScript = true;
+                    providerConfigKey = integration.providerConfigKey;
+                }
+            }
+        }
 
-        const syncInfo = foundConfig
-            ? (foundConfig.syncs || []).find((sync) => sync.name === syncName) || (foundConfig.actions || []).find((action) => action.name === syncName)
-            : null;
+        if ((!syncInfo && !isPostConnectionScript) || !providerConfigKey) {
```
Comment (TBonnin) [inline packages/cli/lib/services/dryrun.service.ts:144]: can we not simply check for `!providerConfigKey`?

**Review by TBonnin — APPROVED** [review]

File: packages/cli/lib/services/dryrun.service.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/cli/lib/services/dryrun.service.ts:144]: it's ts pleasing 😬

**Pull request merged**